lib/bamboo/adapters/local_adapter.ex

```diff
@@ -28,9 +28,22 @@ defmodule Bamboo.LocalAdapter do
     SentEmail.push(email)
   end
 
-  @doc "Adds email to Bamboo.SentEmail. Returns a Task that can be awaited on"
-  def deliver_later(email, _config) do
-    deliver(email, nil)
-    Task.async(fn -> :ok end)
+  def handle_config(config) do
+    case config[:deliver_later_strategy] do
+      nil ->
+        Map.put(config, :deliver_later_strategy, Bamboo.DeliverImmediatelyStrategy)
+      Bamboo.DeliverImmediatelyStrategy ->
+        config
+      _ ->
+        raise ArgumentError, """
```

Comment:
:+1: I'll tackle that sometime this week, unless you want to open a PR :) If you don't have time though that's a-ok